Upgrade to 0.17.53 plus some fixes

hahoyer opened the pull request.

hahoyer committed:
German localization

hahoyer committed:
Merge branch 'master' of https://github.com/judos/hardCrafting

hahoyer committed:
Upgrade to factorio 0.17
Change: flag goes-to-main-inventory removed
Change: Renamed science packs
Change: Upgrade autoplace management, implemented by use of boblib

hahoyer committed:
Next release

hahoyer committed:
Upgrade to factorio 0.17.51
Change: some more german localizations
Change: Renamed science packs
Change: Upgrade autoplace management, implemented by use of boblib

hahoyer committed:
Fork until original mod is upgraded

hahoyer committed:
Fork until original mod is upgraded

hahoyer committed:
Upgrade to factorio 0.17.53
Change: big-processing-machine must not have fast_replacesable_group nor next_upgrade

hahoyer committed:
Bugfixes:
    - fixed problems when patching iron and copper smelting

hahoyer committed:
Bugfixes:
    - technology dependency between electronics and logistics removed

hahoyer committed:
Version: 0.17.1

hahoyer committed:
hotfix: Old version of searchOreAmount was used too

hahoyer committed:
trace: trace removed

hahoyer commented:
Please check discussion tab of hardcrafting2 mod [https://mods.factorio.com/mod/hardCrafting2](url) for another issue

hahoyer committed:
Features:
    - Changed research recipes of pulverizer and big processing machine to be less annoying
    - Improved some locales
  Bugfixes:
    - fixed problems with mod "cargo ships"

hahoyer committed:
Features:
    - Changed research recipes of pulverizer and big processing machine to be less annoying
    - Provide copper-dust production from rich copper ore at big processing machine too
    - Recipes reordered for big processing machine
    - Improved some locales
  Bugfixes:
    - fixed problems with mod "cargo ships"

judos commented:
Hey @hahoyer,
Thanks a lot for the work. I'm checking right now and trying to get an official update out.
As I saw you are more than willing to support mod and I often don't find time to work on it so I added you as collaborator on my repository.
That means you can directly apply changes and we have less merging to do.
I also added you to the permissions on the mod portal. So you are able to push a release if there are bugs or I don't manage to update etc. :)

judos commented on BuildRelease.bat:
```diff
@@ -1,2 +1,3 @@
 java -jar BuildRelease1.2.jar
-pause
\ No newline at end of file
+copy .\release\hardCrafting2_0.17.2.zip d:\Data\Games\factorio\develop\mods
```
@hahoyer I removed these changes to BuildRelease.bat on the official repository because they seem specific to your setup. You can add another bat file and gitignore it if you need these commands ;)

hahoyer commented:
Many thanks,
meanwhile there are some more changes. So if you dont mind I will apply
them to your mod too, later this week.
Greeting Harald


Am Mo., 2. Sept. 2019 um 17:23 Uhr schrieb Julian <notifications@github.com
>:

> Hey @hahoyer <https://github.com/hahoyer>,
> Thanks a lot for the work. I'm checking right now and trying to get an
> official update out.
> As I saw you are more than willing to support mod and I often don't find
> time to work on it so I added you as collaborator on my repository.
> That means you can directly apply changes and we have less merging to do.
> I also added you to the permissions on the mod portal. So you are able to
> push a release if there are bugs or I don't manage to update etc. :)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/judos/hardCrafting/pull/35?email_source=notifications&email_token=AABDICFRGIQ26TWQC4XE3YDQHUVW7A5CNFSM4H5C6SY2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD5WC73Y#issuecomment-527183855>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AABDICDB246HC7IS7XDMFJ3QHUVW7ANCNFSM4H5C6SYQ>
> .
>

judos commented:
Sure, as mentioned feel free to edit. If there are bigger changes or questions we can always discuss on discord or here in tickets. (discord: judos#7996)
